packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -264,6 +281,47 @@ class DateTimePicker extends React.Component {
     }
   }
 
+  handleOnDateInputFocus(event) {
+    this.handleOnInputFocus(event);
+  }
+
+  handleOnTimeInputFocus(event) {
+    this.handleOnInputFocus(event);
+  }
+
+  handleOnInputFocus(event) {
+    if (!this.isDefaultDateAcceptable) {
+      this.dateValue = '';
+      this.timeValue = '';
+      this.handleChange(event, null);
+      this.isDefaultDateAcceptable = true;
+    }
+  }
+
+  handleOnCalendarButtonClick(event) {
```

Comment:
Looking at these examples, should `handleOnCalendarButtonClick` and `handleOnInputFocus` actually be handled by the date-picker themselves? I'm just curious

Reply:
The date-picker and time-input do handle these events to a certain degree. When these events occur, the date-time-picker would also need to manage and update its states and handle the change.